### PR TITLE
Return metadata for underlying collections when generating plots

### LINKED ISSUE #217: bhom/wrapped plot methods: return useful information about the plot
#### Description:

When generating one of these plots, sometimes it is useful to have information about the data behind the plot (time of max temperature as an example). This could simply return some json that has said information contained.

### LINKED ISSUE #219: Python Wind class: prevailing does not ignore calm hours
#### Description:

When running Wind.prevailing() on data in which there are a lot of calm hours, the directions of the calm hours are not ignored when getting the prevailing directions, causing the prevailing direction to not be the same as is apparent in plotted wind rose data.

#### Steps to reproduce:

If using an epw file with bad data, hours with speed=0 don't have directions and are defaulted to 0 degrees (north), which skews prevailing direction to the north, but plotted wind roses show prevailing in a different direction (i.e. south)

#### Expected behaviour:

The prevailing direction only takes into account non-calm hours, and is equivalent to what is plotted on a windrose

#### Test file(s):

Future climate epw files are notoriously bad for wind data, and most of the ones I've seen have missing speeds which are set to 0, and direction to 0 accordingly. The result can be tested with those files.

### LINKED ISSUE #221: Python Wind class: prevailing wind speeds
#### Description:

While working on #217, it's been pointed out that the ability to get wind speeds for the prevailing direction as a method would be incredibly useful (as the implementation is not trivial for a user).

Likely will implement in #220

### LINKED ISSUE #222: Datacollection: mean monthly diurnal ranges
#### Description:

similar to #221, this would be useful to users to get the mean diurnal ranges for each month

### LINKED ISSUE #228: FutureWarning when using SimulationResult with an EPW with no ground temperatures at 0.5m
#### Description:

Warning as below when running a SimulationResult
```
UserWarning: The input EPW doesn't contain any monthly ground temperatures at 0.5m. An approximation method will be used to determine ground temperature at that depth based on ambient dry-bulb.
FutureWarning: 'T' is deprecated and will be removed in a future version, please use 'min' instead.
```

#### Steps to reproduce:

Run a SimulationResult with an epw with no monthly ground temperatures.

#### Expected behaviour:

no deprecated warning.
This can be fixed by replacing line 206 in ladybug_extension/groundtemperature.py 
```
.resample("60T")
```
with 
```
.resample("60min")
```

Also search for any other occurrences where .resample is using "T" or "H" and resolve

### PR DESCRIPTION
**Pull request by Tom-Kingstone**
### Issues addressed by this PR

Closes #217 
Closes #219 
Closes #221 
Closes #222 
Closes #228 

Added metadata gathering methods that provide various important data for an underlying collection/object, and created oM objects that reflect the python metadata dictionaries, along with convert methods in the adapter.

Also fixed an issue where prevailing() did not return the correct wind direction due to calm wind speeds not being incorrectly taken into account.

relevant objects have attribute `NoAutoConstructor`, as they should not be created by anything other than the adapter.

### Test files

[TestPlotCommands.zip](https://github.com/user-attachments/files/16162389/TestPlotCommands.zip)
Test each of the commands, vary inputs to 

### Changelog

* Updated plot Execute methods to return `PlotInformation` instead of a string for the following commands:
    * `HeatPlotCommand`
    * `WindroseCommand`
    * `UTCIHeatPlotCommand`
    * `DiurnalPlotCommand`
    * `SunPathPlotCommand`

### Additional comments

The solar optimisation method will not be being updated like this, as there are about to be changes to the underlying methods, so they will be done afterwards.

### CODE REVIEW

**Review by CKBoulter — APPROVED** [review]
Tested using provided testing file. Successful. Approved!

**Comment by Tom-Kingstone** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by Tom-Kingstone** [comment]
@BHoMBot fix copyright headers ref. 27271951370

**Comment by Tom-Kingstone** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

File: LadybugTools_Engine/Convert/HexCode.cs [hunk 1]
```diff
@@ -30,13 +30,14 @@
...
 
-namespace BH.Engine.LadyBugTools
+namespace BH.Engine.LadybugTools
 {
     public static partial class Convert
     {
         [Description("Converts a colour to its respective RGB hexadecimal code (eg. white => #ffffff).")]
         [Input("colour", "The colour to convert into a hex code.")]
         [Output("hex", "The corresponding hex code.")]
+        [PreviousVersion("7.2", "BH.Engine.LadyBugTools.ToHexCode(System.Drawing.Color)")]
```
Comment (FraserGreenroyd) [inline LadybugTools_Engine/Convert/HexCode.cs:40]:
```suggestion
        [PreviousVersion("7.2", "BH.Engine.LadybugTools.ToHexCode(System.Drawing.Color)")]
```

File: LadybugTools_Engine/Convert/HexCode.cs [hunk 2]
```diff
@@ -30,13 +30,14 @@
...
 
-namespace BH.Engine.LadyBugTools
+namespace BH.Engine.LadybugTools
 {
     public static partial class Convert
     {
         [Description("Converts a colour to its respective RGB hexadecimal code (eg. white => #ffffff).")]
         [Input("colour", "The colour to convert into a hex code.")]
         [Output("hex", "The corresponding hex code.")]
+        [PreviousVersion("7.2", "BH.Engine.LadybugTools.ToHexCode(System.Drawing.Color)")]
```
Comment (FraserGreenroyd) [inline LadybugTools_Engine/Convert/HexCode.cs:40]:
```suggestion
        [PreviousVersion("7.3", "BH.Engine.LadybugTools.ToHexCode(System.Drawing.Color)")]
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check documentation-compliance

File: LadybugTools_Engine/Convert/HexCode.cs [hunk 3]
```diff
@@ -47,6 +48,7 @@ public static string ToHexCode(this Color colour)
         [Description("Converts a string that is in the RGB hexadecimal format into a colour. (eg. #ffffff => white).")]
         [Input("hex", "The hexadecimal representation of a colour.")]
         [Output("colour", "The corresponding colour.")]
+        [PreviousVersion("7.2", "BH.Engine.LadybugTools.FromHexCode(System.String)")]
```
Comment (FraserGreenroyd) [inline LadybugTools_Engine/Convert/HexCode.cs:51]:
```suggestion
        [PreviousVersion("7.3", "BH.Engine.LadybugTools.FromHexCode(System.String)")]
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

File: LadybugTools_Engine/Convert/HexCode.cs [hunk 4]
```diff
@@ -30,13 +30,14 @@
...
 
-namespace BH.Engine.LadyBugTools
+namespace BH.Engine.LadybugTools
 {
     public static partial class Convert
     {
         [Description("Converts a colour to its respective RGB hexadecimal code (eg. white => #ffffff).")]
         [Input("colour", "The colour to convert into a hex code.")]
         [Output("hex", "The corresponding hex code.")]
+        [PreviousVersion("7.3", "BH.Engine.LadybugTools.ToHexCode(System.Drawing.Color)")]
```
Comment (FraserGreenroyd) [inline LadybugTools_Engine/Convert/HexCode.cs:40]:
```suggestion
        [PreviousVersion("7.3", "BH.Engine.LadybugTools.Convert.ToHexCode(System.Drawing.Color)")]
```

File: LadybugTools_Engine/Convert/HexCode.cs [hunk 5]
```diff
@@ -47,6 +48,7 @@ public static string ToHexCode(this Color colour)
         [Description("Converts a string that is in the RGB hexadecimal format into a colour. (eg. #ffffff => white).")]
         [Input("hex", "The hexadecimal representation of a colour.")]
         [Output("colour", "The corresponding colour.")]
+        [PreviousVersion("7.3", "BH.Engine.LadybugTools.FromHexCode(System.String)")]
```
Comment (FraserGreenroyd) [inline LadybugTools_Engine/Convert/HexCode.cs:51]:
```suggestion
        [PreviousVersion("7.3", "BH.Engine.LadybugTools.Convert.FromHexCode(System.String)")]
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Review by FraserGreenroyd — COMMENTED** [review]
@Tom-Kingstone I took a look and tried some things from memory, but it looks like you might need to go for a linked pr in versioning toolkit I'm afraid. Not sure why the previous version isn't working but I no longer have visibility to help debug! Gimme a shout if you want any support or rubber ducking :smile:

**Comment by Tom-Kingstone** [comment]
> @Tom-Kingstone I took a look and tried some things from memory, but it looks like you might need to go for a linked pr in versioning toolkit I'm afraid. Not sure why the previous version isn't working but I no longer have visibility to help debug! Gimme a shout if you want any support or rubber ducking 😄

I was thinking of going that direction too 😄, thanks for the help!

**Comment by Tom-Kingstone** [comment]
@BHoMBot check versioning

**Comment by Tom-Kingstone** [comment]
@BHoMBot check core
@BHoMBot check compliance

**Comment by Tom-Kingstone** [comment]
@BHoMBot check installer

**Review by jamesramsden-bh — DISMISSED** [review]
Code reviewed and tested with supplied test script. A few minor bugs now addressed as per commit history, now all fine.

**Comment by jamesramsden-bh** [comment]
@BHoMBot check required

**Comment by jamesramsden-bh** [comment]
@BHoMBot check compliance

**Comment by jamesramsden-bh** [comment]
@BHoMBot check ready-to-merge

**Comment by jamesramsden-bh** [comment]
@BHoMBot check required

**Comment by Tom-Kingstone** [comment]
@BHoMBot check compliance
@BHoMBot check unit-tests

**Review by jamesramsden-bh — APPROVED** [review]
Fixes have now been tested and working.

**Comment by jamesramsden-bh** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**